### PR TITLE
Feature/node state

### PR DESCRIPTION
**Pull request by minsulee2**
When communicating each other, some node status checks firstly done.

### CODE REVIEW

File: client/index.js [hunk 1]
```diff
@@ -418,6 +418,15 @@ app.get('/get_raw_consensus_state', (req, res) => {
     .end();
 });
 
+
+app.get('/get_node_state', (req, res) => {
```
Comment (liayoo) [inline client/index.js:422]: can we just use the existing /node_status ?

File: p2p/server.js [hunk 2]
```diff
@@ -313,32 +313,36 @@ class P2pServer {
...
-                }
-                if (newTxList.length > 0) {
-                  this.executeAndBroadcastTransaction(
-                      { tx_list: newTxList }, MessageTypes.TRANSACTION);
-                }
-              } else {
-                const createdTx = Transaction.create(tx.tx_body, tx.signature);
+              return;
+            }
+            if (this.node.statue !== BlockchainNodeStatus.SERVING) {
```
Comment (liayoo) [inline p2p/server.js:318]: typo? statue => status?

File: p2p/server.js [hunk 3]
```diff
@@ -313,32 +313,36 @@ class P2pServer {
...
+              logger.debug(`[${LOG_HEADER}] Not ready to cope with transactions.\n` +
```
Comment (liayoo) [inline p2p/server.js:319]: how about accept / process transactions?

File: p2p/server.js [hunk 4]
```diff
@@ -348,6 +352,11 @@ class P2pServer {
             if (this.node.bc.chain.length === 0) {
               return;
             }
+            if (this.node.status !== BlockchainNodeStatus.SERVING) {
+              logger.debug(`[${LOG_HEADER}] Not ready to cope with transactions.\n` +
```
Comment (liayoo) [inline p2p/server.js:356]: ditto

File: client/index.js (hunk 1 above)
Comment (minsulee2) [reply client/index.js]: Oh, I was looking for this. Removed it!

File: p2p/server.js (hunk 2 above)
Comment (minsulee2) [reply p2p/server.js]: Oh, Fixed!

File: p2p/server.js (hunk 3 above)
Comment (minsulee2) [reply p2p/server.js]: I took process here :) thanks!

File: p2p/server.js (hunk 4 above)
Comment (minsulee2) [reply p2p/server.js]: Actually this part is not about transactions but chain segment request. Fixed it Thanks :)

**Review by platfowner — APPROVED** [review]
LGTM
Thanks!

**Review by liayoo — APPROVED** [review]
LGTM, thanks!

**Comment by minsulee2** [comment]
Cheers thank you! merging it now :)

**Pull request merged**